### PR TITLE
🧹 Remove all legacy event dispatchers from CalDAV & CardDAV backends

### PR DESCRIPTION
**Pull request by tcitworld**
Move them to proper EventListeners and test them

### CODE REVIEW

File: apps/dav/lib/Listener/CalendarShareUpdateListener.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+/**
+ * @template-implements IEventListener<\OCA\DAV\Events\CalendarShareUpdatedEvent>
+ */
+class CalendarShareUpdateListener implements IEventListener {
+
+	/** @var Backend */
+	private $activityBackend;
+
+	/** @var LoggerInterface */
+	private $logger;
```
Comment (CarlSchwan) [inline apps/dav/lib/Listener/CalendarShareUpdateListener.php:43]:
```suggestion
	private Backend $activityBackend;
	private LoggerInterface $logger;
```

File: apps/dav/lib/Listener/ClearPhotoCacheListener.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+use OCA\DAV\CardDAV\PhotoCache;
+use OCA\DAV\Events\CardDeletedEvent;
+use OCA\DAV\Events\CardUpdatedEvent;
+use OCP\EventDispatcher\Event;
+use OCP\EventDispatcher\IEventListener;
+
+class ClearPhotoCacheListener implements IEventListener {
+	/** @var PhotoCache */
+	private $photoCache;
```
Comment (CarlSchwan) [inline apps/dav/lib/Listener/ClearPhotoCacheListener.php:36]:
```suggestion
	private PhotoCache $photoCache;
```

**Review by CarlSchwan — APPROVED** [review]
So nice :heart_eyes:

File: apps/dav/lib/Listener/SubscriptionListener.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @copyright 2021 Thomas Citharel <nextcloud@tcit.fr>
```
Comment (ChristophWurst) [inline apps/dav/lib/Listener/SubscriptionListener.php:6]: happy new year :)

**Review by ChristophWurst — APPROVED** [review]
:broom: 

https://geek-and-poke.com/geekandpoke/2017/2/18/50-shades-part-2

**Review by miaulalala — APPROVED** [review]
so-fresh-clean

So fresh and so clean!

**Comment by CarlSchwan** [comment]
the ci failures about the acceptance tests are unrelated (some issues with npm and a recent changes)

**Pull request merged**